feat: Grep table valued function (#188)
askgitdev/askgit: README.md

#### Utilities

> ##### GREP

Reviewer: ```suggestion
##### grep
```

> Table valued function that searches on a string

Reviewer: ```suggestion
A table-valued function that searches text by line for a string match.
Meant to behave somewhat similarly to the `grep` command line tool.
```

| Column      | Type     |
|-------------|----------|
| line_no     | INT      |
| line        | TEXT     |

```SQL

> SELECT * FROM grep(<contents>,<search_string>,<preceeding_lines>,<proceeding_lines>)
> SELECT * FROM grep(<contents>,<search_string>) -- preceeding_lines and proceeding_lines default to 0

Reviewer: ```suggestion
SELECT * FROM grep(<contents>, <search_string>, <before>, <after>)
SELECT * FROM grep(<contents>, <search_string>) -- before and after default to 0
```
